platform/vcs-impl/src/com/intellij/openapi/vcs/annotate/ExtendedActiveAnnotationGutter.java

```diff
@@ -0,0 +1,22 @@
+// Copyright 2000-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
+package com.intellij.openapi.vcs.annotate;
+
+import com.intellij.openapi.util.Couple;
+import com.intellij.openapi.vcs.actions.ActiveAnnotationGutter;
+import com.intellij.openapi.vcs.history.VcsRevisionNumber;
+
+import java.awt.*;
+import java.util.Map;
+
+public interface ExtendedActiveAnnotationGutter extends ActiveAnnotationGutter {
```

Comment:
This interface looks unnecessary complicated.
It also exposes implementation details.